scripts/builtin/mark_down_creator.py

```diff
@@ -0,0 +1,385 @@
+
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+# This python script does take all files within the current directory and creates a corresponding markdown file out of the existing headers
+#   If there are specific parts of the header missing a warning within the terminal will be displayed
+#   The finished markdown file is then placed in the same directory
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+# OUTPUTS
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+#   Name                            Type    Default     Meaning
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+#   SystemDs_Builtin_Markdown       .dm     -------     A .dm file of all files with a parsable header
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+# Additional Information
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+# If the output within the terminal does not look correct because the os isnt supporting colored output,
+#   please swap the comment prefix in line 300 and 301
+
+
+
+
+#This Array contains all valid information
+file_data_array = []
+
+#This Array contains all invalid file names with corresponding missing values
+incorrect_file_data = []
+
+#Global FLAG to identify if Headers with only 3 arguments are valid or not!
+ALLOW_DIFFERENT_SIZED_HEADERS = False
+
+
+
+#This class contains all necessary variables for the complete representation of the markdown file
+class File_data:
+    def __init__(self, inputParameterCount, inputParams, I_HEADER_MISSING, I_DESCRIPTION_MISSING, outputParameterCount, outputParams, O_HEADER_MISSING, O_DESCRIPTION_MISSING, description, additional, fileName):
+        self.param_count = inputParameterCount
+        self.param = inputParams
+        self.input_header_missing = I_HEADER_MISSING
+        self.input_description_missing = I_DESCRIPTION_MISSING
+        self.output_param_count = outputParameterCount
+        self.output = outputParams
+        self.output_header_missing = O_HEADER_MISSING
+        self.output_description_missing = O_DESCRIPTION_MISSING
+        self.description = description
+        self.additional = additional
+        self.fileName = fileName
+
+#class to make a pretty output possible
+class bcolors:
+    HEADER = '\033[95m'
```

Comment:
On windows, unfortunately, only these color strings are output as is - it would be great to make this more robust or go back to basic output formatting.